plugin/qbackendconnection.cpp

```diff
@@ -402,122 +399,67 @@ void QBackendConnection::setState(ConnectionState newState)
         Q_ASSERT(oldState == ConnectionState::WantTypes);
         if (m_qmlEngine) {
             // immediately transition
-            setState(ConnectionState::WantRoot);
+            setState(ConnectionState::Ready);
         } else {
             qCDebug(lcConnection) << "State -- Got types. Want engine.";
         }
         break;
-    case ConnectionState::WantRoot:
-        Q_ASSERT(oldState == ConnectionState::WantEngine);
-        qCDebug(lcConnection) << "State -- Got engine, want root.";
-        break;
-    case ConnectionState::Established:
+    case ConnectionState::Ready:
         Q_ASSERT(m_qmlEngine);
-        Q_ASSERT(oldState == ConnectionState::WantRoot);
-        // ok, we have ROOT, types, and an engine. try process it all.
+        Q_ASSERT(oldState == ConnectionState::WantEngine);
         qCDebug(lcConnection) << "State -- Entered established state. Flushing pending.";
-        handlePendingMessages();
         break;
+    default:
+        Q_UNREACHABLE();
```

Comment:
I'd suggest not adding a default case. That way, when adding additional states, it's easy to find places where it's missing.

If you do have one, avoid Q_UNREACHABLE unless you are certain it cannot ever be hit. Things get very unhappy if you end up actually reaching those at runtime, and they are usually unpleasant to debug. Maybe use Q_ASSERT_X or something instead.